### PR TITLE
MSI - Onboarding AWS OSD workflows

### PR DESCRIPTION
**Pull request by omrirh**

### CODE REVIEW

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse ci/rehearse/CSPI-QE/MSI/rhods-operator/v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Review by oharan2 — COMMENTED** [review]
Maybe change the title from AWS/IPI to OSD/IPI :)

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-qe-test-addon-v4.13-qetest-addon-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-qe-test-addon-v4.13-qetest-addon-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-addon-interop-v4.13-GA-stage-smoke-osd-rhods-addon-baseline-fips

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-qe-test-addon-v4.13-qetest-addon-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-addon-interop-v4.13-GA-stage-smoke-osd-rhods-addon-baseline-fips

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-qe-test-addon-v4.13-qetest-addon-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-ipi-osd-qetest-operator-aws-ipi

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-qe-test-addon-v4.13-qetest-addon-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-osd-qetest-operator-aws-osd

**Comment by omrirh** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-addon-interop-v4.13-GA-stage-smoke-osd-rhods-addon-baseline-fips

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-qe-test-addon__v4.13.yaml [hunk 1]
```diff
@@ -21,19 +21,13 @@ resources:
       cpu: 100m
       memory: 200Mi
 tests:
-- as: qetest-rosa-addon
+- as: qetest-addon-aws-osd
```
Comment (rnetser) [inline ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-qe-test-addon__v4.13.yaml:24]: please create a separate file; do not er-use this one

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-operator__v4.13-aws-osd.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      version: "4.13"
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: qetest-operator-aws-osd
```
Comment (rnetser) [inline ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-operator__v4.13-aws-osd.yaml:28]: wrong name

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-operator__v4.13-aws-osd.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  steps:
+    cluster_profile: aws-managed-cspi-qe
+    env:
+      COMPUTE_NODES: "6"
+      OPENSHIFT_VERSION: "4.13"
+      OPERATOR1_CONFIG: name=rhods-operator;namespace=redhat-ods-operator
+    workflow: operator-aws-osd
```
Comment (rnetser) [inline ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-operator__v4.13-aws-osd.yaml:35]: no test is running, why do we need this?

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+workflow:
+  as: addon-aws-osd
+  steps:
+    env:
+      API_HOST: stage
+      CHANNEL_GROUP: candidate
+      COMPUTE_MACHINE_TYPE: "m5.4xlarge"
+      MACHINE_POOL_INSTANCE_TYPE: "m5.2xlarge"
```
Comment (rnetser) [inline ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml:8]: why are these 2 needed?

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      COMPUTE_NODES: "3"
+      PARALLEL: "true"
+      MULTI_AZ: "true"
+      DISABLE_WORKLOAD_MONITORING: "false"
```
Comment (rnetser) [inline ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml:12]: is this relevant for osd?

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      - ref: osd-ccs-cluster-provision-aws
+      - ref: osd-ccs-conf-machine-pool
+      - ref: osd-ccs-conf-idp-htpasswd
+      - ref: osd-ccs-conf-idp-htpasswd-multi-users
+      - ref: osd-ccs-cluster-operators-wait-ready
+      - ref: addon-install
+    post:
+      - ref: osd-ccs-cluster-deprovision
+  documentation: |-
+    This workflow installs an OSD cluster with the specified configures using the lastest nightly build.
```
Comment (rnetser) [inline ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml:27]: not nightly but what is provided in OPENSHIFT_VERSION

File: ci-operator/step-registry/addons-and-operators/aws/osd/addons-and-operators-aws-osd-workflow.yaml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+workflow:
```
Comment (rnetser) [inline ci-operator/step-registry/addons-and-operators/aws/osd/addons-and-operators-aws-osd-workflow.yaml:1]: same comments / questions

File: ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+workflow:
```
Comment (rnetser) [inline ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml:1]: ditto

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-operator__v4.13-aws-osd.yaml (hunk 3 above)
Comment (omrirh) [reply ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-operator__v4.13-aws-osd.yaml:35]: it's for sanity check on osd operators flow

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-qe-test-addon__v4.13.yaml (hunk 1 above)
Comment (omrirh) [reply ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-qe-test-addon__v4.13.yaml]: done

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-operator__v4.13-aws-osd.yaml (hunk 2 above)
Comment (omrirh) [reply ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-operator__v4.13-aws-osd.yaml]: changed to `sanity-aws-osd-rhods-operator`.
let me know if that's ok

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml (hunk 4 above)
Comment (omrirh) [reply ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml]: removed cluster infra settings from all workflows.
we will provide that from outside as the scenario requires.

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml (hunk 6 above)
Comment (omrirh) [reply ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml]: done

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml (hunk 5 above)
Comment (omrirh) [reply ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml]: no, it is false by default anyway.
thanks for pointing out

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-qe-test-addon-v4.13-aws-osd-sanity-aws-osd-rhods-addon

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-rhods-operator-v4.13-aws-osd-sanity-aws-osd-rhods-operator

**Comment by omrirh** [comment]
/pj-rehearse periodic-ci-CSPI-QE-MSI-rhods-addon-interop-v4.13-GA-stage-smoke-osd-rhods-addon-baseline-fips

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-qe-test-addon-v4.13-aws-osd-sanity-aws-osd-rhods-addon

**Comment by omrirh** [comment]
/pj-rehearse pull-ci-CSPI-QE-MSI-qe-test-addon-v4.13-aws-osd-sanity-aws-osd-rhods-addon

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  as: addon-aws-osd
+  steps:
+    env:
+      API_HOST: stage
+      CHANNEL_GROUP: candidate
+      PARALLEL: "true"
+      MULTI_AZ: "true"
+      ENABLE_AUTOSCALING: "false"
+      MIN_REPLICAS: "3"
+      MAX_REPLICAS: "6"
```
Comment (rnetser) [inline ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml:11]: why do we need this and not just `REPLICAS`?

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+      ETCD_ENCRYPTION: "true"
+    pre:
+      - ref: osd-ccs-cluster-provision-aws
+      - ref: osd-ccs-conf-machine-pool
+      - ref: osd-ccs-conf-idp-htpasswd
+      - ref: osd-ccs-conf-idp-htpasswd-multi-users
+      - ref: osd-ccs-cluster-operators-wait-ready
+      - ref: addon-install
+    post:
+      - ref: osd-ccs-cluster-deprovision
```
Comment (rnetser) [inline ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml:21]: please add firewatch

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-addon-interop__v4.13-GA-stage.yaml [hunk 11]
```diff
@@ -44,6 +44,26 @@ tests:
...
+      COMPUTE_MACHINE_TYPE: g4dn.2xlarge
+      COMPUTE_NODES: "2"
+      FIPS: "true"
+      MULTI_AZ: "false"
+      OPENSHIFT_VERSION: "4.13"
+      OPERATOR1_CONFIG: name=gpu-operator-certified;namespace=nvidia-gpu-operator;source=certified-operators;target-namespaces=nvidia-gpu-operator,
+      OPERATOR2_CONFIG: name=servicemeshoperator
+      OPERATOR3_CONFIG: name=serverless-operator;namespace=openshift-serverless
+      OPERATOR4_CONFIG: name=openshift-pipelines-operator-rh;namespace=openshift-pipelines;channel=latest
+      OPERATOR5_CONFIG: name=nfd;namespace=openshift-nfd;target-namespaces=openshift-nfd,
```
Comment (rnetser) [inline ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-addon-interop__v4.13-GA-stage.yaml:62]: firewatch config is missing

File: ci-operator/step-registry/addons-and-operators/aws/osd/addons-and-operators-aws-osd-workflow.yaml [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    pre:
+      - ref: osd-ccs-cluster-provision-aws
+      - ref: osd-ccs-conf-machine-pool
+      - ref: osd-ccs-conf-idp-htpasswd
+      - ref: osd-ccs-conf-idp-htpasswd-multi-users
+      - ref: osd-ccs-cluster-operators-wait-ready
+      - ref: operator-install
+      - ref: addon-install
+    post:
+      - ref: osd-ccs-cluster-deprovision
```
Comment (rnetser) [inline ci-operator/step-registry/addons-and-operators/aws/osd/addons-and-operators-aws-osd-workflow.yaml:22]: ditto

File: ci-operator/step-registry/addons-and-operators/aws/osd/addons-and-operators-aws-osd-workflow.yaml [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+  as: addons-and-operators-aws-osd
+  steps:
+    env:
+      API_HOST: stage
+      CHANNEL_GROUP: candidate
+      PARALLEL: "true"
+      MULTI_AZ: "true"
+      ENABLE_AUTOSCALING: "false"
+      MIN_REPLICAS: "3"
+      MAX_REPLICAS: "6"
```
Comment (rnetser) [inline ci-operator/step-registry/addons-and-operators/aws/osd/addons-and-operators-aws-osd-workflow.yaml:11]: ditto

File: ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+  as: operator-aws-osd
+  steps:
+    env:
+      API_HOST: stage
+      CHANNEL_GROUP: candidate
+      PARALLEL: "true"
+      MULTI_AZ: "true"
+      ENABLE_AUTOSCALING: "false"
+      MIN_REPLICAS: "3"
+      MAX_REPLICAS: "6"
```
Comment (rnetser) [inline ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml:11]: ditto

File: ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+      ETCD_ENCRYPTION: "true"
+    pre:
+      - ref: osd-ccs-cluster-provision-aws
+      - ref: osd-ccs-conf-machine-pool
+      - ref: osd-ccs-conf-idp-htpasswd
+      - ref: osd-ccs-conf-idp-htpasswd-multi-users
+      - ref: osd-ccs-cluster-operators-wait-ready
+      - ref: operator-install
+    post:
+      - ref: osd-ccs-cluster-deprovision
```
Comment (rnetser) [inline ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml:21]: ditto

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml (hunk 9 above)
Comment (omrirh) [reply ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml]: removed; not using autoscaling so it's redundant

File: ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml (hunk 10 above)
Comment (omrirh) [reply ci-operator/step-registry/addon/aws/osd/addon-aws-osd-workflow.yaml:20]: done

File: ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-addon-interop__v4.13-GA-stage.yaml (hunk 11 above)
Comment (omrirh) [reply ci-operator/config/CSPI-QE/MSI/CSPI-QE-MSI-rhods-addon-interop__v4.13-GA-stage.yaml:86]: done

File: ci-operator/step-registry/addons-and-operators/aws/osd/addons-and-operators-aws-osd-workflow.yaml (hunk 12 above)
Comment (omrirh) [reply ci-operator/step-registry/addons-and-operators/aws/osd/addons-and-operators-aws-osd-workflow.yaml:21]: done

File: ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml (hunk 14 above)
Comment (omrirh) [reply ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml]: done

File: ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml (hunk 15 above)
Comment (omrirh) [reply ci-operator/step-registry/operator/aws/osd/operator-aws-osd-workflow.yaml:20]: done

**Pull request merged**